guide: add Configuration guide (#4379)
iterative/dvc.org: content/docs/user-guide/project-structure/configuration.md

# DVC Configuration

Once initialized in a <abbr>project</abbr>, DVC populates its installation
directory with [internal files], which include `.dvc/config`, the default
configuration file.

[internal files]: /doc/user-guide/project-structure/internal-files

<admon type="warn">

> `.dvc/config` is meant to be tracked by Git and should not contain sensitive
> user info or secrets (passwords, SSH keys, etc). Other
> [config file locations](#config-file-locations) can be used as well.

Reviewer: ```suggestion
`.dvc/config` is meant to be tracked by Git and should not contain sensitive
user info or secrets (passwords, SSH keys, etc). Other
[config file locations](#config-file-locations) are not tracked by Git and can
be used for sensitive info.
```

Minor suggestion to be more direct in this explanation.

Reviewer: Or use what's in the ref: `Use --local when in doubt`.

Author: Hmmm, didn't love either suggestion. I updated it in a different way (b7e9672ff), PTAL.

Reviewer: What's your goal with this admon? I was trying to make sure that we include enough info so people know where they generally should store sensitive info. I feel like just saying that other locations can be used kind of leaves people hanging.

Author: Good point. We could move it into the next section (**Config file locations**)...

Author: Done too.

Reviewer: Looks like you pushed to #4380 instead of this PR?

Reviewer: Anyway, looks good there, thanks!

within:

- [`core`](#core) - main section with the general config options
- [`remote`](#remote) - sections in the config file that describe [remote
  storage]
- [`cache`](#cache) - options that affect the project's <abbr>cache</abbr>
- [`exp`](#exp) - options to change the default repo paths assumed by
  `dvc exp init`
- [`hydra`](#hydra) - options around [Hydra Composition] for experiment
  configuration.
- [`parsing`](#parsing) - options around the parsing of [dictionary unpacking].
- [`plots`](#plots) - options for configuring `dvc plots`.
- [`state`](#state) - see [Internal directories and files][internals] to learn
  more about the state database.
- [`index`](#index) - see [Internal directories and files][internals] to learn
  more about remote index files.

[remote storage]: /doc/user-guide/data-management/remote-storage
[hydra composition]: /doc/user-guide/experiment-management/hydra-composition
[dictionary unpacking]:
  /doc/user-guide/project-structure/dvcyaml-files#dictionary-unpacking
[internals]: /doc/user-guide/project-structure/internal-files

### core

- [`core.remote`](#remote) - name of the default remote storage

> - `core.interactive` - whether to always ask for confirmation before reproducing

Author: I just wonder whether these should be hidden in `<details>` sections, so the page doesn't seem that long.

Reviewer: Have you tried L2 headers instead of L3 so they all show in the sidebar?

Author: I guess that would also be better than now, yes. I'd still also consider using hidden sections... Maybe both?

Author: U: I tried combining both and it does work (when you click in the title on the right sidebar it scrolls and opens the section). Up to you @dberenbaum

Author: p.s. updated to use both for now, PTAL.
